src/main/java/com/google/api/codegen/util/ViewNamer.java

```diff
@@ -0,0 +1,75 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util;
+
+/**
+ * A ViewNamer provides names for specific components of a view.
+ *
+ * Naming is composed of two steps:
+ *
+ * 1. Composing a Name instance with the name pieces
+ * 2. Formatting the Name for the particular type of identifier needed.
+ *
+ * This class delegates step 2 to the provided name formatter, which generally
+ * would be a language-specific namer.
+ */
+public class ViewNamer implements NameFormatter {
```

Comment:
I don't really think it would be "much simpler." For example, SurfaceNamer would implement NameFormatter (just like NameFormatterMixin now does) and forward calls to NameFormatter, which I think is one aspect that might be considered confusing about the current design, but which doesn't really change. It would just end up making SurfaceNamer a little bigger in order to move the NameFormatterMixin methods into it. I know there is no imminent sister class of SurfaceNamer, but conceivably the Discovery code might want to create such a class if they decide to use the MVVM pattern.